scripts/poms/sbndpoms_caf_concatenation.sh

```diff
@@ -0,0 +1,290 @@
+    FLATFILE="$WORKDIR/$FLATNAME"
+    FLATJSONFILE="${FLATFILE}.json"
+    flatten_caf $OUTDIR/$UNIQUEOUTCONCATNAME $FLATFILE
+
+    extractCAFMetadata "$FLATFILE" > "$FLATJSONFILE"
+
+    # Steal the ifdh and sam commands from sbndpoms_genfclwithrunnumber_maker.sh (Thanks Dom)
+    # SAM needs the file to have a unique name
+    ifdh addOutputFile $FLATFILE
+    ifdh renameOutput unique
+
+    # Bit annoying but we now need to find the fcl file again as ifdh doesn't tell us what the unique name is
+    if [[ `find $WORKDIR -name "${FLATNAME%.*}*.root" | wc -l` -ne 1 ]]
+    then
+      echo "Found incorrect number of matching files for pattern: ${FLATNAME%.*}*.root"
+      find $WORKDIR -name "${FLATNAME%.*}*.root"
+      echo "Exiting"
+      exit 3
+    else
+      UNIQUEOUTFLATNAME=`find $WORKDIR -name "${FLATNAME%.*}*.root"`
+      UNIQUEOUTFLATNAME=`basename $UNIQUEOUTFLATNAME`
+      echo "$FLATNAME renamed to $UNIQUEOUTFLATNAME"
+    fi
+
+    # OK so it looks like there it exactly one pattern match, so assume that is the correct one
+    #Copy the file to the output directory (most likely dcache)
+    ifdh copyBackOutput $OUTDIR
+    #Clear up
+    ifdh cleanup
+
+    sbndpoms_metadata_extractor.sh "$WORKDIR/$UNIQUEOUTFLATNAME"
+
+    samweb -e sbnd declare-file "$FLATJSONFILE"
+    samweb -e sbnd add-file-location ${UNIQUEOUTFLATNAME} $OUTDIR
+    echo "$UNIQUEOUTFLATNAME declared and located by SAM"
+  fi
+}
+
+prepare
+
+# Loop over all of the files in the dataset
+DEFSIZE=$(samweb -e sbnd list-definition-files --summary $DEFNAME | grep "Total size" | tr -dc '0-9')
+FILECOUNT=$(samweb -e sbnd list-definition-files --summary $DEFNAME | grep "File count" | tr -dc '0-9')
+
+# Work out the size of each file
+FILESIZE=$(( $DEFSIZE / $FILECOUNT ))
+# Work out the max number of file to concat together
+SLICELIMIT=$(( $MAXSIZE / $FILESIZE ))
+# Work out the number of concats this will result in
+CONCATCOUNT=$(( $FILECOUNT / $SLICELIMIT ))
+# Spread the files evenly between the CONCATS
+FILELIMIT=$(( $FILECOUNT / $CONCATCOUNT ))
+FILELIMIT=$(( FILELIMIT + 1 ))
+
+echo "DefSize: $DEFSIZE and FileCount: $FILECOUNT"
+echo "FileSize: $FILESIZE and FileLimit: $FILELIMIT"
+echo "ConcatCount: $CONCATCOUNT"
+
+while [[ $SLICENUM -lt $CONCATCOUNT ]]
+do
+  FILECOUNTER=$(( $SLICENUM * $FILELIMIT ))
+  echo "Creating Slice: $SLICENUM, starting from file: $FILECOUNTER"
+
+  SLICEDEFNAME=${DEFNAME}_Slice${SLICENUM}
+  samweb -e sbnd create-definition "$SLICEDEFNAME" "defname: $DEFNAME with limit $FILELIMIT with offset $FILECOUNTER"
+
+  doConcat
+
+  SLICENUM+=1
+done
+
+# Create a definition with the output files
+samweb -e sbnd create-definition $MDPRODUCTIONDEFNAME "file_name like concat_caf_%.root and file_type $MDFILETYPE and production.type $MDPRODUCTIONTYPE and production.name $MDPRODUCTIONNAME and sbnd_project.name $MDSBNDPROJECTNAME and sbnd_project.version $MDSBNDPROJECTVERSION and sbnd_project.stage $MDSBNDPROJECTSTAGE"
```

Comment:
You could throw in `is_child_of: $DEFNAME` (not 100% sure of the syntax). I'm kind of paranoid about this sort of definition going wrong.